src/nvim/drawscreen.c

```diff
@@ -715,14 +716,17 @@ void end_search_hl(void)
   screen_search_hl.rm.regprog = NULL;
 }
 
-static void win_redr_bordertext(win_T *wp, VirtText vt, int col)
+static void win_redr_bordertext(win_T *wp, VirtText vt, int col, BorderTextType bt)
 {
   for (size_t i = 0; i < kv_size(vt);) {
     int attr = 0;
     char *text = next_virt_text_chunk(vt, &i, &attr);
     if (text == NULL) {
       break;
     }
+    if (attr == 0) {
```

Comment:
if attr was _explicitly_ set to zero by including a zero in the tuple like `[["item", 0], ...]` this must be respected.